src/rhsm/connection.py

```diff
@@ -1141,7 +1137,15 @@ def _request(
 
         self.validateResponse(result, request_type, handler)
 
-        return result
+        # Handle 204s
+        if not len(result["content"]):
+            return None
+        try:
+            return json.loads(result["content"])
+        except json.JSONDecodeError:
+            # This is primarily intended for getting releases from CDN, because
+            # the file containing releases is plaintext and not json.
+            return result["content"]
```

Comment:
I guess that we should not solve removing of `RestLib` class using this way and return only content. At least doc string and type-hints do not correspond to the state of code.

Maybe it would be better to return dictionary with result and return `result["content"]` from methods like `request_get()`, `request_post()`... Maybe we should return `result["content"]` later and we should return content from callers of `request_get()`... like from methods of `UEPConnection` class e.g. `getConsumer()`.

It seems that you assume that dictionary `result` is returned. Look at method `unregisterConsumer()`, but it is not true ATM.

Reply:
Yes, that would be a better approach. I'll check it out. Thanks for your feedback.